### PR TITLE
feat: add path to mutator function

### LINKED ISSUE #92: Add Option to have mutator function provide path of subschema
**Is your feature request related to a problem? Please describe.**
Validator requires the path of the subschema when while traversing the schema to be validated. 

**Describe the solution you'd like**
Traverse to pass the path of the current subschema as a prop to the mutator function. Could be implemented as a traverse option.

### PR DESCRIPTION
**Pull request by mattbford**
fixes #92

### CODE REVIEW

**Comment by mattbford** [comment]
TODO: needs array implementation and then it should be good to go

File: src/index.ts [hunk 1]
```diff
@@ -67,6 +81,7 @@ export default function traverse(
   traverseOptions = defaultOptions,
   depth = 0,
   recursiveStack: JSONSchema[] = [],
+  pathStack: any[] = [],
```
Comment (BelfordZ) [inline src/index.ts:84]:
```suggestion
  pathStack: string[] = [],
```

File: src/index.ts [hunk 2]
```diff
@@ -42,6 +44,10 @@ export const defaultOptions: TraverseOptions = {
   bfs: false,
 };
 
+const jsonPathStringify = (s: any[]) => {
```
Comment (BelfordZ) [inline src/index.ts:47]: should this be `s: string[]`?

File: src/index.ts [hunk 3]
```diff
@@ -123,29 +134,58 @@ export default function traverse(
...
+    pathStack.pop();
   } else {
     let itemsIsSingleSchema = false;
 
     if (schema.items) {
       if (schema.items instanceof Array) {
-        mutableSchema.items = schema.items.map(rec);
+        pathStack.push("items");
+        mutableSchema.items = schema.items.map((x,i) => {
+          pathStack.push(`${i}`);
```
Comment (BelfordZ) [inline src/index.ts:176]: i.toString(); is preferable over implicit type cast

**Review by BelfordZ — CHANGES_REQUESTED** [review]
looks great, just a couple nits.

File: src/index.ts [hunk 4]
```diff
@@ -84,7 +84,7 @@ export default function traverse(
   // thus adding it to the recursive stack will prevent it from being explored if the
   // boolean is seen in a further nested schema.
   if(depth === 0) {
```
Comment (BelfordZ) [inline src/index.ts:86]:
```suggestion
  if (depth === 0) {
```

hmm linting should have caught this...

**Review by BelfordZ — APPROVED** [review]

**Pull request merged**